File: sources/com/s8/pkgs/people/menu/P8MenuIcon.java
package com.s8.pkgs.people.menu;

import com.s8.api.web.S8WebFront;
import com.s8.api.web.S8WebObject;
import com.s8.pkgs.people.WebSources;


/**
 * 
 */
public class P8MenuIcon extends S8WebObject {

	
	public static P8MenuIcon create(S8WebFront front) {
		P8MenuIcon menu = new P8MenuIcon(front);
		return menu;
	}
	
	
	public P8MenuIcon(S8WebFront front) {
		super(front, WebSources.ROOT_PATH + "/menu/P8MenuIcon");
	}
	
	
	public void setImage(String url) {
		vertex.outbound().setStringUTF8Field("imageURL", url);
	}
	
}

File: web-sources/S8-pkgs-people/menu/P8MenuIcon.js

import { S8 } from "/S8-api/S8Context.js";
import { S8Object } from "/S8-api/S8Object.js";

import { S8WebFront } from "/S8-pkgs-ui-carbide/S8WebFront.js";


/**
 * 
 */
S8WebFront.CSS_import('/S8-pkgs-people/menu/P8Menu.css');


/**
 * 
 */
export class P8MenuIcon extends S8Object {


    /**
     * 
     */
    constructor(){
        super();
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("p8menu-icon");

        S8.page.SVG_insertByPathname(this.wrapperNode, "/S8-pkgs-people/icons/person-circle.svg", 32, 32);
    }


    /**
     * 
     * @returns 
     */
    getEnvelope(){
        return this.wrapperNode;
    }


    /**
     * icon
     * @param {*} code 
     */
    S8_set_imageURL(path){
        this.wrapperNode.style.backgroundImage = `url(${path})`;
    }

   /** continuous rendering approach */
    S8_render(){}

    /** continuous rendering approach */
    S8_dispose(){}
}